lib/storage/src/content_manager/consensus_state.rs

```diff
@@ -614,6 +630,14 @@ impl Persistent {
         self.save()
     }
 
+    pub fn last_applied_entry(&self) -> Option<u64> {
+        if self.snapshot_was_applied {
+            Some(self.state.hard_state.commit)
+        } else {
+            self.unapplied_entries.get_last_applied()
```

Comment:
**unapplied**_entries.get_last_**applied** looks counter-intuitive. Maybe we should come with better name for `unapplied_entries`?